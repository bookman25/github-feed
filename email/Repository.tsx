import * as React from 'react';
import { IRepositoryConfig } from '../query/query';
import Commit, { ICommit } from './Commit';
import Milestones, { IMilestone } from './Milestones';
import Releases, { IRelease } from './Releases';

export interface IProps {
	repo: IRepositoryConfig;
	commit?: ICommit;
	releases?: IRelease[];
	milestones?: IMilestone[];
}

export default function Repository({ repo, commit, releases, milestones }: IProps) {
	return <div>
		<h2><a href={ `https://github.com/${repo.user}/${repo.repo}` }>{ repo.repo }</a></h2>
		<Commit {...commit} />
		<Releases releases={ releases } />
		<Milestones milestones={ milestones } />
	</div>;
}