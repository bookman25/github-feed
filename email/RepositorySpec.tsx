jest.unmock('./Repository');

import * as React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Repository from './Repository';

describe('Respository template', () => {
	it('repository name when no other data', () => {
		const view = shallow(<Repository repo={ { repo: 'name', user: 'user' } } />);
		expect(shallowToJson(view)).toMatchSnapshot();
	});
});