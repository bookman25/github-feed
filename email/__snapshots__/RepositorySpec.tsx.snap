exports[`Respository template repository name when no other data 1`] = `
<div>
  <h2>
    <a
      href="https://github.com/user/name">
      name
    </a>
  </h2>
  <Commit />
  <Release
    releases={undefined} />
  <Milestones
    milestones={undefined} />
</div>
`;
